osu.Game/Rulesets/Mods/ModDaycore.cs

```diff
@@ -27,11 +27,11 @@ protected ModDaycore()
             }, true);
         }
 
-        public override void ApplyToTrack(Track track)
+        public override void ApplyToTrack(ITrack track)
         {
             // base.ApplyToTrack() intentionally not called (different tempo adjustment is applied)
-            track.AddAdjustment(AdjustableProperty.Frequency, freqAdjust);
-            track.AddAdjustment(AdjustableProperty.Tempo, tempoAdjust);
+            (track as IAdjustableAudioComponent)?.AddAdjustment(AdjustableProperty.Frequency, freqAdjust);
+            (track as IAdjustableAudioComponent)?.AddAdjustment(AdjustableProperty.Tempo, tempoAdjust);
```

Comment:
It feels a little clumsy that `ITrack` is not an `IAdjustableAudioComponent`, even more so seeing that `ITrack` even has [a specific method for removing speed adjustments](https://github.com/ppy/osu-framework/blob/05c118dbc94fd2e21e71595f8cf62631c94efe15/osu.Framework/Audio/Track/ITrack.cs#L78-L81).

Reply:
I agree, but I'm not sure what the best solution is to this. I'm not 100% sure `ITrack` should be an `IAdjustableAudioComponent` and would like more thoughts on that, with the obvious alternative being to remove `ResetSpeedAdjustments` from `ITrack`.

@peppy any opinion?

Reply:
maybe just nuke the method and use the full interface here..

Reply:
So as it turns out the method isn't actually coming from `ITrack`, it's coming from `IAdjustableClock`... `ITrack` only duplicates it.

I went with the solution proposed above, which is to make `ITrack` an `IAdjustableAudioComponent`.